embed/ephy-embed.c

```diff
@@ -957,3 +958,28 @@ ephy_embed_inspector_is_loaded (EphyEmbed *embed)
 
   return embed->inspector_loaded;
 }
+
+void
+ephy_embed_attach_notification_manager (EphyEmbed *embed)
+{
+  EphyNotificationManager *manager;
+
+  g_return_if_fail (EPHY_IS_EMBED (embed));
+
+  manager = ephy_notification_manager_dup_singleton ();
+  gtk_overlay_add_overlay (GTK_OVERLAY (embed->overlay), GTK_WIDGET (manager));
```

Comment:
I'm commenting as I go, so I haven't looked at your implementation of EphyNotificationManager yet, but I'm really surprised that it's a GtkWidget. I've never seen a singleton GtkWidget before; this doesn't seem like a good idea.